src/cdk-experimental/testing/testbed/testbed-harness-environment.ts

```diff
@@ -14,7 +14,7 @@ import {UnitTestElement} from './unit-test-element';
 
 /** A `HarnessEnvironment` implementation for Angular's Testbed. */
 export class TestbedHarnessEnvironment extends HarnessEnvironment<Element> {
-  protected constructor(rawRootElement: Element, private _fixture: ComponentFixture<unknown>) {
+  constructor(rawRootElement: Element, private _fixture: ComponentFixture<unknown>) {
```

Comment:
@mmalerba I removed this since for the snack bar test environment we need to get harnesses from the `document.body > .cdk-overlay-container` and using the fixture element as root therefore does not work.

Reply:
I think @crisbeto ran into this in a different PR and we decided that `getDocumentRoot` is implemented wrong and it should return `document.body` instead. That way you can use `this.documentRootLocatorFactory()` to accomplish this in your harness. His PR probably just didn't get merged yet

Reply:
I have it in the autocomplete PR. I think that I’ll split it out into a separate PR so we aren’t blocked on other overlay-based components.

Reply:
Awesome. I'll base that PR on top of @crisbeto's changes.

Reply:
@mmalerba Just tried that out but unfortunately it's not sufficient for the snack-bar harness. This is because snack-bar's are never load-able as part of the fixture but rather the harness needs to be loaded from the document root.

e.g.

```ts
// pseudo-code
TestBedHarnessEnvironment.loader(fixture)
  .documentRootLocator().getHarness(SnackBarHarness)
```

I'm not sure what's the ideal solution. Maybe the loader should have an option to get harnesses from the document root?

Reply:
Ahh interesting, so I guess this is only a problem for the `TestbedHarnessEnvironment` since it uses the fixture's root element as the root of the `HarnessLoader`. The protractor one uses `document.body` which works for overlays too.

The solutions I would propose are:
1. Just make `TestbedHarnessEnvironment.loader` use `document.body` instead of the fixture root as its root. This would make it just work like it does in protractor, but I'm not sure, are there potential bad consequences of doing this? I'm not sure what else is usually in the DOM in a unit test, outside of the fixture.
2. Make a separate static method like `TestbedHarnessEnvironment.documentRootLoader`. Most of the time people would just use `loader` but for special cases like snackbar they could use `documentRootLoader`. I don't love this, I'm ok if the harness author has to know their component is special (`documentRootLoactorFactor`), I don't like it when the test author has to know though...

Reply:
@mmalerba Yeah... that feels like a difficult question. I'd personally go with (2) as (1) is simply not explicit and definitely could lead to side-effects as there could be left-overs in the document body.

Also totally agree with your concerns in option (2). how should we proceed? leave this like that for now and discuss this more or go with one of the options?

Reply:
I would actually lean towards the first option instead of adding another API. If we have problems with something being left in the DOM after tests, it's almost always overlay-related anyway.

Reply:
It's true that most of the time this is either an overlay, the aria-describer element, aria-live announcer element, but considering that these harnesses are also outside of this project, we don't know what libraries are used, or whether tests are always cleaned up properly.

This can cause a *lot* of confusion and take up a lot of time (finding the root cause of the leftover etc.). I don't feel too strong about it though. Happy to go with (1) if desired.